### PR TITLE
Example Pose Estimation Network with SE3 loss function

### PR DESCRIPTION
**Pull request by farrell236**

### CODE REVIEW

File: examples/se3_pose_estimation/make_dataset_KingsCollege.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
```
Comment (johmathe) [inline examples/se3_pose_estimation/make_dataset_KingsCollege.py]: This should probably live in the example directory. It's not really a minimal example

File: examples/se3_pose_estimation/make_dataset_KingsCollege.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+# ================================================================================
```
Comment (johmathe) [inline examples/se3_pose_estimation/make_dataset_KingsCollege.py]: We don't define the coding and use ==== as a syntax in geomstats

File: examples/se3_pose_estimation/make_dataset_KingsCollege.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+# Original
+# File: create_posenet_lmdb_dataset.py
+# Link: https://git.io/fpPuw
+# Author: Alex Kendall <https://alexgkendall.com>
+#
+# Modified
+# File: make_dataset_Kingscollege.py
```
Comment (johmathe) [inline examples/se3_pose_estimation/make_dataset_KingsCollege.py]: no caps in filenames

File: examples/se3_pose_estimation/make_dataset_KingsCollege.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+import random
+
+import numpy as np
+import tensorflow as tf
+
+from tqdm import tqdm
+from geomstats.special_orthogonal_group import SpecialOrthogonalGroup
+
+
+###############################################################################
```
Comment (johmathe) [inline examples/se3_pose_estimation/make_dataset_KingsCollege.py]: avoid this comment style

File: examples/se3_pose_estimation/make_dataset_KingsCollege.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    return tf.train.Feature(int64_list=tf.train.Int64List(value=[value]))
+
+def _float_feature(value):
+    return tf.train.Feature(float_list=tf.train.FloatList(value=[value]))
+
+
+###############################################################################
+# Dataset Locations
+
+directory   = '/vol/medic01/users/bh1511/4_CVPR2019/data_raw/KingsCollege/'
```
Comment (johmathe) [inline examples/se3_pose_estimation/make_dataset_KingsCollege.py]: that looks very specific to cambridg'es infrastructure

File: examples/se3_pose_estimation/make_dataset_KingsCollege.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+poses   = []
+images  = []
+
+with open(directory+dataset) as f:
+    next(f)  # skip the 3 header lines
+    next(f)
+    next(f)
+    for line in f:
+        fname, p0, p1, p2, p3, p4, p5, p6 = line.split()
+        p0 = float(p0)
```
Comment (johmathe) [inline examples/se3_pose_estimation/make_dataset_KingsCollege.py]: poses = [float(p) for p in line.split()]

File: examples/se3_pose_estimation/make_dataset_KingsCollege.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+
+###############################################################################
+# Process Lables
+
+poses   = []
+images  = []
+
+with open(directory+dataset) as f:
+    next(f)  # skip the 3 header lines
```
Comment (johmathe) [inline examples/se3_pose_estimation/make_dataset_KingsCollege.py]: for loop?

File: examples/se3_pose_estimation/make_dataset_KingsCollege.py [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+    pose_q  = np.array(poses[i][3:7])
+    pose_x  = np.array(poses[i][0:3])
+
+    rot_vec = SO3_GROUP.rotation_vector_from_quaternion(pose_q)[0]
+    pose    = np.concatenate((rot_vec, pose_x), axis=0)
+
+    X = imageio.imread(images[i])
+    X = X[::4, ::4, :]
+    #X = exposure.equalize_hist(X)
```
Comment (johmathe) [inline examples/se3_pose_estimation/make_dataset_KingsCollege.py]: remove leftover comment

File: examples/se3_pose_estimation/make_dataset_KingsCollege.py [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+        'pose':     _bytes_feature(pose_raw),
+        'pose_q':   _bytes_feature(pose_q_raw),
+        'pose_x':   _bytes_feature(pose_x_raw)}))
+
+    writer.write(example.SerializeToString())
+
+writer.close()
+
+
+print('\n', 'Creating Dataset Success.')
```
Comment (johmathe) [inline examples/se3_pose_estimation/make_dataset_KingsCollege.py]: can you flake8 all this? there should be a main and a function in this file

File: examples/se3_pose_estimation/test_se3_KingsCollege.py [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+                #'height':      tf.FixedLenFeature([], tf.int64),
+                #'width':       tf.FixedLenFeature([], tf.int64),
+                'image':        tf.FixedLenFeature([], tf.string),
+                'pose':         tf.FixedLenFeature([], tf.string)
+            })
+
+        image = tf.decode_raw(features['image'], tf.uint8)
+        pose = tf.decode_raw(features['pose'], tf.float32)
+
+        #height = tf.cast(features['height'], tf.int32)
```
Comment (johmathe) [inline examples/se3_pose_estimation/test_se3_KingsCollege.py]: leftover comments

File: examples/se3_pose_estimation/train_se3_KingsCollege.py [hunk 11]
```diff
@@ -0,0 +1,165 @@
+# -*- coding: utf-8 -*-
+# ================================================================================
```
Comment (johmathe) [inline examples/se3_pose_estimation/train_se3_KingsCollege.py]: ditto

avoid caps

**Comment by johmathe** [comment]
Any update on this?

**Comment by ninamiolane** [comment]
Merged in applications repository
